python/src/scipp/compat/mantid.py

```diff
@@ -261,6 +288,30 @@ def convertEventWorkspace_to_dataarray(ws, load_pulse_times):
     return sc.DataArray(**coords_labs_data)
 
 
+def convertMDHistoWorkspace_to_dataset(md_histo):
+    ndims = md_histo.getNumDims()
+    if ndims > 3:
+        raise RuntimeError("Converter cannot process md histo workspace input "
+                           "with > 3 dimensions. Input has {}.".format(ndims))
+    coords = dict()
+    dims_used = []
+    for i in range(ndims):
+        dim = md_histo.getDimension(i)
+        frame = dim.getMDFrame()
+        sc_dim = md_dimension(dim, i)
+        coords[sc_dim] = sc.Variable(dims=[sc_dim],
+                                     values=np.linspace(
+                                         dim.getMinimum(), dim.getMaximum(),
+                                         dim.getNBins()),
+                                     unit=md_unit(frame))
+        dims_used.append(sc_dim)
+    data = sc.Variable(dims=dims_used,
+                       values=md_histo.getSignalArray(),
+                       variances=md_histo.getErrorSquaredArray())
+    nevents = sc.Variable(dims=dims_used, values=md_histo.getNumEventsArray())
+    return sc.DataArray(coords=coords, data=data, attrs={'nevents': nevents})
```

Comment:
Is it worth extracting data from the `ExperimentInfo`s that the base class stores? Could reuse most of the code from handling `MatrixWorkspace`. I think this is not urgently required, so maybe better to postpone.

Reply:
Good idea. I'll look into that.